src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1287 @@
+    });
+    if(lightCount) arrayAppend(fields, SceneFieldData{
+        SceneField::Light, lightObjects, lights, SceneFieldFlag::OrderedMapping
+    });
+    if(cameraCount) arrayAppend(fields, SceneFieldData{
+        SceneField::Camera, cameraObjects, cameras, SceneFieldFlag::OrderedMapping
+    });
+
+    /* Convert back to the default deleter to avoid dangling deleter function
+       pointer issues when unloading the plugin */
+    arrayShrink(fields, DefaultInit);
+
+    return SceneData{SceneMappingType::UnsignedInt, nodeCount, std::move(data), std::move(fields)};
+}
+
+SceneField UfbxImporter::doSceneFieldForName(Containers::StringView name) {
+    for(UnsignedInt i = 0; i < Containers::arraySize(sceneFieldNames); ++i) {
+        if(name == sceneFieldNames[i]) return sceneFieldCustom(i);
+    }
+    return SceneField{};
+}
+
+Containers::String UfbxImporter::doSceneFieldName(UnsignedInt name) {
+    if(name < Containers::arraySize(sceneFieldNames))
+        return sceneFieldNames[name];
+    return {};
+}
+
+UnsignedLong UfbxImporter::doObjectCount() const {
+    return _state->objectCount;
+}
+
+Long UfbxImporter::doObjectForName(const Containers::StringView name) {
+    const ufbx_scene* scene = _state->scene.get();
+    const ufbx_node* node = ufbx_find_node_len(scene, name.data(), name.size());
+    return node ? Long(node->typed_id - _state->nodeIdOffset) : -1;
+}
+
+Containers::String UfbxImporter::doObjectName(const UnsignedLong id) {
+    /* Should always be in bounds as AbstractImporter validates the index */
+    return _state->scene->nodes[id + _state->nodeIdOffset]->name;
+}
+
+UnsignedInt UfbxImporter::doCameraCount() const {
+    return UnsignedInt(_state->scene->cameras.count);
+}
+
+Int UfbxImporter::doCameraForName(const Containers::StringView name) {
+    return typedId(ufbx_find_element_len(_state->scene.get(), UFBX_ELEMENT_CAMERA, name.data(), name.size()));
+}
+
+Containers::String UfbxImporter::doCameraName(const UnsignedInt id) {
+    return _state->scene->cameras[id]->name;
+}
+
+Containers::Optional<CameraData> UfbxImporter::doCamera(UnsignedInt id) {
+    const ufbx_camera* camera = _state->scene->cameras[id];
+
+    if(camera->projection_mode == UFBX_PROJECTION_MODE_PERSPECTIVE) {
+        return CameraData{CameraType::Perspective3D,
+            Deg(Float(camera->field_of_view_deg.x)), Float(camera->aspect_ratio),
+            Float(camera->near_plane), Float(camera->far_plane)};
+    } else if(camera->projection_mode == UFBX_PROJECTION_MODE_ORTOGRAPHIC) {
+        return CameraData{CameraType::Orthographic3D,
+            Vector2(camera->orthographic_size),
+            Float(camera->near_plane), Float(camera->far_plane)};
+    } else {
+        /* Not expected to gain new projection modes */
+        /* LCOV_EXCL_START */
+        Error() << "Trade::UfbxImporter::camera(): camera projection mode" << camera->projection_mode << "is not supported";
+        return {};
+        /* LCOV_EXCL_STOP */
```

Comment:
This is a bit janky, ufbx is not expected to ever get more projection modes but I'm not sure how defensive we want to be with such things.

Reply:
> ufbx is not expected to ever get more projection modes

In that case I'd put this into a switch without the `default:` case, so in case it ever does, the compiler notifies me with a warning, and an unreachable assert right after:

```cpp
    switch(camera->projection_mode) {
        case UFBX_PROJECTION_MODE_PERSPECTIVE:
            return CameraData{CameraType::Perspective3D,
                Deg(Float(camera->field_of_view_deg.x)), Float(camera->aspect_ratio),
                Float(camera->near_plane), Float(camera->far_plane)};
        case UFBX_PROJECTION_MODE_ORTOGRAPHIC:
            return CameraData{CameraType::Orthographic3D,
                Vector2(camera->orthographic_size),
                Float(camera->near_plane), Float(camera->far_plane)};
    };
    
    /* Not expected to gain new projection modes */
    CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
}
```